lib/castle/client.rb

```diff
@@ -6,27 +6,39 @@ class Client
 
     def initialize(request, options = {})
       @do_not_track = default_tracking(options)
-      cookies = options[:cookies] || request.cookies
-      client_id = Extractors::ClientId.new(request, cookies).call('__cid')
+      request_headers = Extractors::Headers.new(request).call
       ip = Extractors::IP.new(request).call
-      headers = Extractors::Headers.new(request).call
-      @api = API.new(client_id, ip, headers)
+      @context = setup_context(request_headers, ip, options[:context])
+      @castle_headers = setup_castle_headers(request, request_headers, ip, options[:cookies])
+      @api = API.new(@castle_headers)
     end
 
-    def fetch_review(id)
-      @api.request_query("reviews/#{id}")
+    def fetch_review(review_id)
+      @api.request_query("reviews/#{review_id}")
     end
 
-    def identify(args)
-      @api.request('identify', args) if tracked?
+    def identify(event, options = {})
+      return unless tracked?
+      command = Castle::Commands::Identify.new(@context).build(event, options || {})
+      @api.request(command)
     end
 
-    def authenticate(args)
-      @api.request('authenticate', args)
+    def authenticate(event, user_id, options = {})
+      return unless tracked?
```

Comment:
The context from options should be merge with `@context` but should only be used for this request and not persisted for any following requests. https://paper.dropbox.com/doc/Server-SDK-specs-5WNJdpiGefPk2I1aAtWET#:uid=074686716191575160848076&h2=Context-Merging